server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/QueryExecution.java

```diff
@@ -478,10 +478,19 @@ private ExecutionResult getExecutionResult(QueryState state) {
 
     // collect redirect info to client for writing
     if (analysis.getStatement() instanceof InsertBaseStatement) {
+      if (state != QueryState.FINISHED) {
+        //if insert not finished, then before redirect to client: clean partition cache and re analyse to get the current
+        // region locations, so the redirect node is available. maybe region migrated when writing.
+        partitionFetcher.invalidAllCache();
```

Comment:
This will cause drastic decrease of writing and reading performance. You should invalid specific cache entries instead of all cache.